sklearn/preprocessing/_encoders.py

```diff
@@ -474,10 +474,15 @@ def inverse_transform(self, X):
             # drop will either be None or handle_unknown will be error. If
             # self.drop is not None, then we can safely assume that all of
             # the nulls in each column are the dropped value
-            elif self.drop is not None:
+            else:
                 dropped = np.asarray(sub.sum(axis=1) == 0).flatten()
                 if dropped.any():
-                    X_tr[dropped, i] = self.categories_[i][self.drop_idx_[i]]
+                    if self.drop is None:
+                        raise ValueError("Unknown value in row {}".format(i))
```

Comment:
`i` is the feature index not the row. The rows are in `np.flatnonzero(dropped)`. The message can be more explicit:

> The follow samples do not have an inverse when drop=None and handle_unknown='error' because they are all zeros: [0, 1, 2].